### ISSUE TITLE
[Transition] MaterialContainerTransform IndexOutOfBoundsException

### ISSUE THREAD

**Issue by wesam-al-tabbaa** [material-components/material-components-android#2217]
When implementing a shared-element transition via MaterialContainerTransform between two Fragments, I'm encountering the following Exception:

    java.lang.IndexOutOfBoundsException: Index: 4, Size: 4
        at java.util.ArrayList.get(ArrayList.java:437)
        at androidx.fragment.app.FragmentTransitionImpl.setNameOverridesReordered(FragmentTransitionImpl.java:183)
        at androidx.fragment.app.DefaultSpecialEffectsController.startTransitions(DefaultSpecialEffectsController.java:665)
        at androidx.fragment.app.DefaultSpecialEffectsController.executeOperations(DefaultSpecialEffectsController.java:114)
        at androidx.fragment.app.SpecialEffectsController.executePendingOperations(SpecialEffectsController.java:294)
        at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2202)
        at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2106)
        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:2002)
        at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:524)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:224)
        at android.app.ActivityThread.main(ActivityThread.java:7590)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)

One important note is that this issue only appeared after updating to AndroidX Fragment 1.3.4 and I suspect the issue might have something to do with the following patch in that version:

"When doing a shared element transition on a ViewGroup, if the ViewGroup has transitionGroup set to false, it will now properly transition"
https://android-review.googlesource.com/q/I99675eac030325415789be0762aa666355f27dd8

It seems that shared elements across the 2 Fragments are being counted incorrectly, and using `transitionGroup` to control the number of shared elements had no effect at all.

The transition is from a list item in Fragment 1 to Fragment 2

list item layout

```
<com.google.android.material.card.MaterialCardView
    android:id="@+id/card_view_email_root"
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:layout_marginHorizontal="@dimen/margin_container"
    android:layout_marginTop="@dimen/margin_container"
    android:onClick="@{onClickListener}"
    android:transitionName="@{String.valueOf(email.id)}">

    <LinearLayout
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:orientation="horizontal"
        android:padding="@dimen/margin_container">

        <com.google.android.material.textview.MaterialTextView
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:layout_marginEnd="@dimen/margin_form"
            android:text="@{String.valueOf(email.id)}"
            android:textStyle="bold"
            tools:text="1" />

        <com.google.android.material.textview.MaterialTextView
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:text="@{email.subject}"
            tools:text="Email 1" />

    </LinearLayout>

</com.google.android.material.card.MaterialCardView>
```

Fragment 2 layout

```
<LinearLayout
    android:id="@+id/linear_layout_email_details_root"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:orientation="vertical"
    android:padding="@dimen/margin_container"
    android:transitionName="@string/transition_name_email_details">

    <com.google.android.material.textview.MaterialTextView
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_marginBottom="@dimen/margin_form"
        android:text="@{String.valueOf(email.id)}"
        android:textStyle="bold"
        tools:text="1" />

    <com.google.android.material.textview.MaterialTextView
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_marginBottom="@dimen/margin_form"
        android:text="@{email.subject}"
        tools:text="Email 1" />

    <com.google.android.material.textview.MaterialTextView
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_marginBottom="@dimen/margin_form"
        android:text="@{email.body}"
        tools:text="text" />

    <com.google.android.material.textview.MaterialTextView
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:text="@{email.date}"
        tools:text="10 / 10 / 2010" />

</LinearLayout>
```

**Comment by vpuonti** [comment]
https://github.com/material-components/material-components-android/issues/2215#issuecomment-847033187

Using `FragmentManager.enableNewStateManager(false)` seems to fix the issue. 

Tested on 
```
"androidx.appcompat:appcompat:1.3.0"
"androidx.fragment:fragment-ktx:1.3.4"
"com.google.android.material:material:1.3.0"
```

**Comment by wesam-al-tabbaa** [comment]
> [#2215 (comment)](https://github.com/material-components/material-components-android/issues/2215#issuecomment-847033187)
> 
> Using `FragmentManager.enableNewStateManager(false)` seems to fix the issue.
> 
> Tested on
> 
> ```
> "androidx.appcompat:appcompat:1.3.0"
> "androidx.fragment:fragment-ktx:1.3.4"
> "com.google.android.material:material:1.3.0"
> ```

Yeah but I believe such a solution is not a proper one. A patch should be issued to ensure compatibility between the two libraries instead of sacrificing all the benefits of the new Fragment state manager.

**Comment by HappyDr0id** [comment]
@wesam-al-tabbaa It looks like this issue has been solved in `1.3.5`: https://developer.android.com/jetpack/androidx/releases/fragment#1.3.5
